lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -88,28 +96,18 @@ def add_included(resource_name, serializers, parent = nil)
 
         def attributes_for_serializer(serializer, options)
           if serializer.respond_to?(:each)
-            result = []
-            serializer.each do |object|
-              result << resource_object_for(object, options)
-            end
+            serializer.map { |s| resource_object_for(s, options) }
```

Comment:
I'm working on the final bit to improve handling of the "included"
relationships. After that, it should be pretty clean/understandable/easy to
work with.

Le mardi 1 septembre 2015, João Moura <notifications@github.com> a écrit :

> In lib/active_model/serializer/adapter/json_api.rb
> <https://github.com/rails-api/active_model_serializers/pull/1103#discussion_r38415770>
> :
>
> > @@ -88,28 +96,18 @@ def add_included(resource_name, serializers, parent = nil)
> >
> >          def attributes_for_serializer(serializer, options)
> >            if serializer.respond_to?(:each)
> > -            result = []
> > -            serializer.each do |object|
> > -              result << resource_object_for(object, options)
> > -            end
> > +            serializer.map { |s| resource_object_for(s, options) }
>
> [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails-api/active_model_serializers/pull/1103/files#r38415770>
> .
>


-- 
Lucas Hosseini
lucas.hosseini@gmail.com